worker/worker/request_handlers.go

```diff
@@ -489,3 +515,87 @@ func (s StreamContext) getStreamNameVoD() string {
 	}
 	return s.courseSlug
 }
+
+// extractKeywords creates images of a given stream in a temporary folder, extracts keywords, sends the keywords to
+// TUM-Live, and deletes the temporary folder.
+func extractKeywords(ctx *StreamContext) error {
+	log.WithField("File", ctx.getTranscodingFileName()).Println("Start extracting keywords")
+
+	// create temporary directory
+	dir, err := ioutil.TempDir("", "keyword-extraction")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+	defer os.Remove("out.jpeg") // Remove thumbgen's out.jpeg
+
+	// generate images to process
+	g, err := thumbgen.New(ctx.getTranscodingFileName(), 768, 128,
+		"out.jpeg", thumbgen.WithJpegCompression(70), thumbgen.WithStoreSingleFrames(dir))
```

Comment:
Using the same filename for every stream will likely cause trouble with parallel processing of multiple lectures

Reply:
Changed to `fmt.Sprintf("%s.jpeg", uuid.NewUUID())`